### DISCUSSION TITLE
Applying for a 7590 device

### DISCUSSION THREAD

**Discussion by xaldyz** [PeterPawn/modfs#35] [Handling problems / questions]
Hello Peter,
I started to look into enabling the SSH access for this router and I almost instantly came across your posts. 
They are very  helpful and full of information, especially [this](https://www.ip-phone-forum.de/threads/fritz-box-7580-firmware-153-06-90-telnet-service-freischalten-geht-auch-für-7560-und-7590.296678/) page (I don't speak German so I used google translate)!
However, I'm not able to understand how to apply this project upon an existing, out-of-the-box 7590, with OS 7.21, firmware 154.07.21, maybe just for the language problem :)
Since it's my only router, it would be unfortunate if I wasn't able to recover it, so I am quite scared into applying firmware modifications.
I've already tried the "# 96 * 7 *" method, and it's not working (although the "# 96 * 8 *" responds).
I am not a computer novice, and I am confident into kernel compilation and similar stuff, but the "embedded" part of it (so bootloader, image into RAM through EVA, ...) is not crystal clear for me. Also, some commands are updated, I can pack a step-by-step English instruction for newcomers like me.

I tried to upload the modfs.tgz image to the "file FRITZ!OS" but it didn't work, the system gave me with "incorrect file" and then rebooted with factory settings.
Do I have to build a YourFritz kernel image and flash it?
It doesn't seem so in the [aforementioned post](https://www.ip-phone-forum.de/threads/fritz-box-7580-firmware-153-06-90-telnet-service-freischalten-geht-auch-für-7560-und-7590.296678/), but I don't get the section about the "switch system", and the "bootloader not working after a soft reset".

However, I had a peek at eva_switch_system (which uses eva_discover) but the system is not able to find the FritzBox, maybe because you need to power-cycle the device in the meanwhile? But it doesn't seem to work either.
By hard-wiring the IP address in the variable EVA_IP, the function eva_switch_interface is able to connect, but it fails since the "SYST" command returns "UNIX Type: L8 Version: Linux 4.9.198", and the script is complaining that it should be "AVM whatever".
I use Ubuntu 18.04 in WSL (version 1), I don't know if it may be the source of the problem.

Maybe this issue does not belong here and is more suited to YourFritz, but since I "only" want to enable Telnet, I tought this repository is the most minimal configuration.

I can provide more information if those are helpful, I put everything that seemed relevant.
Can you help me out? 
Thank you!

**Comment by PeterPawn** [comment]
Better ('cause more recent) source for modifications to non-VR9-models: 

https://www.ip-phone-forum.de/threads/wie-modifiziere-ich-die-originale-firmware-vom-hersteller-und-wie-installiere-ich-meine-eigene-dann-auf-der-fritz-box.307161/

You should try first, to get more experiences, how to catch a FRITZ!Box device in its bootloader (EVA) - you're the only one, who may define/decide, whether you're able to connect to the FTP server (from EVA) for sure, if you need it sometimes in the future. And you will need it, if you're trying a modified firmware and it will not start/run without problems ... one of the basic tasks in this case is switching to the previous system via the other `linux_fs_start` value and this may be set from FTP server best.

First try the recovery software by AVM - it removes any existing settings, but you should even take an otherwise unused device for your exercises. But it installs the original firmware, too, and is another option, if settings AND firmware have to be changed/replaced.

If you see the mentioned response for a `SYST` command to the (FTP) server, you're connected with the (full-featured) FTP server of a running FRITZ!OS system and not to the (rudimentary) version from the EVA loader.

And by the way ... an attempt to upload the `modfs.tgz` file to the box (either from the update page and/or the (EVA) FTP server) can't be any serious action. This is a (zipped) TAR file containing shell scripts and some binaries - it's simply (in this form) not intended to be uploaded to a FRITZ!OS device.

Please try to read (and understand) the thread from the link above - meanwhile I'll convert this `Issue` to a `Discussion`, 'cause it's more a usage problem than an error of the provided files/logic/samples.
